### PR TITLE
clap-utils: Fix signer resolution on Windows

### LINKED ISSUE #17252: Path not support hidden folder. OS:Windows
#### Problem
OS: windows

Below command has an error after 1.6.8, But fine on 1.5.X
```
solana-keygen new
solana-keygen pubkey
* Error: unrecognized signer source
```

I change to the below command. It work fine. 
```
solana-keygen new -o 123.json
solana-keygen pubkey 123.json
* AtpiSfWpGcd19mvQS8qMLuvJNta5g6qUfxtff1s3XvaU
```

#### Proposed Solution
Could path support hidden folder like previous version (.config)

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

`SignerSource` in `clap-utils` can't resolve Windows paths

#### Summary of Changes

* Burn a day on getting a reliable Windows build env up
* Fix it

fixes #17252

### CODE REVIEW

File: clap-utils/src/keypair.rs [hunk 1]
```diff
@@ -211,7 +211,17 @@ pub(crate) fn parse_signer_source<S: AsRef<str>>(
...
     let source = source.as_ref();
-    match uriparse::URIReference::try_from(source) {
+    let source = {
+        #[cfg(target_family = "windows")]
+        {
+            source.replace("\\", "/")
+        }
+        #[cfg(not(target_family = "windows"))]
+        {
+            source.to_string()
```
Comment (CriesofCarrots) [inline clap-utils/src/keypair.rs:221]: Pity we have to do this

**Review by CriesofCarrots — DISMISSED** [review]
lgtm, just one place I think would benefit from a comment

File: clap-utils/src/keypair.rs (hunk 1 above)
Comment (t-nelson) [reply clap-utils/src/keypair.rs:221]: Yeah :face_with_head_bandage:

File: clap-utils/src/keypair.rs [hunk 2]
```diff
@@ -231,18 +241,22 @@ pub(crate) fn parse_signer_source<S: AsRef<str>>(
                         legacy: false,
                     }),
                     SIGNER_SOURCE_STDIN => Ok(SignerSource::new(SignerSourceKind::Stdin)),
-                    _ => Err(SignerSourceError::UnrecognizedSource),
+                    _ => {
+                        #[cfg(target_family = "windows")]
+                        if scheme.len() == 1 {
```
[the remark this answers is not in the document]
Comment (t-nelson) [reply clap-utils/src/keypair.rs:246]: Good call!

**Pull request merged**